src/crypto/ripemd160.h

```diff
@@ -28,18 +29,11 @@ class CRIPEMD160
 };
 
 /** Compute the 160-bit RIPEMD-160 hash of an array. */
-inline uint160 RipeMd160(const unsigned char* data, size_t size)
+inline uint160 RipeMd160(Span<const unsigned char> data)
 {
     uint160 result;
-    CRIPEMD160().Write(data, size).Finalize(result.begin());
+    CRIPEMD160().Write(data.data(), data.size()).Finalize(result.begin());
     return result;
 }
 
-/** Compute the 160-bit RIPEMD-160 hash of an object. */
-template <typename T>
-inline uint160 RipeMd160(const T& container)
```

Comment:
Instead of adding code and removing it in the next commit, it might be better to squash the commits. This will reduce review burden for reviewers starting from scratch as well as reduce the git history, blame depth and clone size.